components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/dao/OAuthAppDAO.java

```diff
@@ -947,17 +973,60 @@ private boolean isUsernameCaseSensitive(String tenantQualifiedUsername) {
      * @return
      * @throws IdentityOAuth2Exception
      */
+    /**
+     * @deprecated use {@link #getOIDCIdTokenAudiences(String, String )} instead.
+     */
+    @Deprecated
     public List<String> getOIDCAudiences(String tenantDomain, String consumerKey) throws IdentityOAuth2Exception {
 
-        List<String> audiences = new ArrayList<>();
+        return this.getOIDCIdTokenAudiences(tenantDomain, consumerKey);
+
+    }
+
+    /**
+     * Retrieves OIDC ID Token audience values configured for an oauth consumer app.
+     *
+     * @param tenantDomain application tenant domain
+     * @param consumerKey  client ID
+     * @return
+     * @throws IdentityOAuth2Exception
+     */
+    public List<String> getOIDCIdTokenAudiences(String tenantDomain, String consumerKey)
+            throws IdentityOAuth2Exception {
+
+        List<String> idTokenAudiences = new ArrayList<>();
+        idTokenAudiences = getAudiencesFromDB(idTokenAudiences, tenantDomain, consumerKey);
+        return idTokenAudiences;
+    }
+
+
+    /**
+     * Retrieves OIDC Access Token audience values configured for an oauth consumer app.
+     *
+     * @param tenantDomain application tenant domain
+     * @param consumerKey  client ID
+     * @return
+     * @throws IdentityOAuth2Exception
+     */
+    public List<String> getOIDCAccessTokenAudiences(String tenantDomain, String consumerKey)
+            throws IdentityOAuth2Exception {
+
+        List<String> accessTokenAudiences = new ArrayList<>();
+        accessTokenAudiences = getAudiencesFromDB(accessTokenAudiences, tenantDomain, consumerKey);
+        return accessTokenAudiences;
+    }
+
+    private List<String> getAudiencesFromDB(List<String> audiences, String tenantDomain, String consumerKey)
+            throws IdentityOAuth2Exception {
+
         Connection connection = IdentityDatabaseUtil.getDBConnection(false);
         PreparedStatement prepStmt = null;
         ResultSet rSetAudiences = null;
         try {
             prepStmt = connection.prepareStatement(SQLQueries.OAuthAppDAOSQLQueries.GET_SP_OIDC_PROPERTY);
             prepStmt.setString(1, consumerKey);
             prepStmt.setInt(2, IdentityTenantUtil.getTenantId(tenantDomain));
-            prepStmt.setString(3, OPENID_CONNECT_AUDIENCE);
+            prepStmt.setString(3, OPENID_CONNECT_ACCESS_TOKEN_AUDIENCE);
```

Comment:
is it always this in both cases?

Reply:
no. Resolved with d43a24390e23624cc0139558087860229e9dbe7e